projects/packages/stats-admin/src/class-rest-controller.php

```diff
@@ -120,27 +140,47 @@ public function check_user_privileges_callback() {
 	}
 
 	/**
-	 * Resource endpoint.
+	 * Stats resource endpoint.
 	 *
 	 * @param WP_REST_Request $req The request object.
 	 * @return array
 	 */
-	public function get_stats_resource_from_wpcom( $req ) {
-		if ( 'file-downloads' === $req->get_param( 'resource' ) ) {
-			return $this->empty_result();
+	public function get_stats_resource( $req ) {
+		switch ( $req->get_param( 'resource' ) ) {
+			case 'file-downloads':
+			case 'video-plays':
+			case 'clicks':
+			case 'search-terms':
+			case 'top-authors':
+			case 'country-views':
+			case 'referrers':
+			case 'top-posts':
+			case 'publicize':
+			case 'followers':
+			case 'tags':
+			case 'visits':
+			case 'comments':
+			case 'comment-followers':
+			case 'streak':
+			case 'insights':
+			case 'highlights':
+				return static::request_as_blog_cached(
```

Comment:
It's so handy just to send the requests here without the `WPCOM_Stats` lib.

Reply:
TBH I'd still create separate endpoint in order to properly validate and sanitize request args.
However, if you prefer to keep it like this, why not do something like:

```
case 'file-downloads':
  return $this->wpcom_stats->get_file_downloads( $req->get_params() );
case 'video-plays':
   return $this->wpcom_stats->get_video_plays( $req->get_params() );
```

Reply:
> However, if you prefer to keep it like this, why not do something like:
case 'file-downloads': return $this->wpcom_stats->get_file_downloads( $req->get_params() );

Addressed in b3fd2cb0db